Panoramic controls for 360 videos

boyaquito opened the pull request:
Hello,

I wanted to share this since years ago. Please let me know if it can be useful.

Thanks,

Benjamin-Dobell committed:
Added LineSegments object support to ObjectLoader. (#9334)

mkeblx committed:
Raycaster docs html fix (#9341)

Forgot the br.

Benjamin-Dobell committed:
Added Object3D.quaternion support to ObjectLoader. (#9335)

dhritzkiv committed:
DRYer texture creation for different map types (#9336)

* DRYer texture creation for different map types

Noticed there was a fair bit of repetition in each map line case. This pulls the case logic out into a reusable function

This also adds support for texture parameters for specular maps as a result. (Expands on #9311 and #9308)

* Change style of function declaration

Commit:
include packing in points_frag.glsl to fix compile error (#9355)

spidersharma03 committed:
Bloom Pass - Inspired from Unreal Engine (#9351)

* Bloom Pass Inspired from Unreal
* Extract Bright Pass
* Mip Map blur Pass
* Combine Pass

* resize function in Bloom

resize function in Bloom

* Bug Fixes and a few Changes

* Resize bug fix
* changed blur filter. Need to reinvestigate, the older blur sometime
* Changed kernel size. was too big.

Rich-Harris committed:
Modules (#9310)

* convert to ES modules

* rebuild

* move shaders back into glsl files

* reinstate polyfills

mrdoob committed:
ShaderChunk: Removed unused ambient_pars.

mrdoob committed:
Updated builds.

mrdoob committed:
Removed unused variables.

mrdoob committed:
Updated builds.

mrdoob committed:
src/audio clean up

mrdoob committed:
BufferAttribute clean up.

mrdoob committed:
Fixed function formatting

Rich-Harris committed:
Remove unused `is*` properties (#9381)

* move is* properties to prototypes and remove unused ones

* rebuild

* merge dev -> less-is

* remove rebuild from PR

Rich-Harris committed:
Update CONTRIBUTING.md (#9382)

This will hopefully prevent other people making the same mistake as me (https://github.com/mrdoob/three.js/pull/9381#issuecomment-234625456)

GGAlanSmithee committed:
Update ISSUE_TEMPLATE.md (#9385)

mrdoob committed:
Updated builds.

Rich-Harris committed:
remove comments and unnecessary newlines from glsl files (#9389)

mrdoob committed:
Updated builds.

karimbeyrouti committed:
Added `setWithCredentials` function to: ImageLoader, and TextureLoader which send the `withCredentials` variable to the XHRLoader. (#9391)

takahirox committed:
Clean up WebGLRenderer (#9406)

Commit:
Update ExtrudeGeometry.js (#9409)

Changed the bevel so that it is elliptical instead of sinusoidal. Formatted both areas the same with the linear option in the comments.

spidersharma03 committed:
Outline Pass Algorithm (#9407)

1. Draw Non Selected objects in the depth buffer
2. Make non selected objects invisible, and draw only the selected objects, by comparing the depth buffer of non selected objects. Write a mask here for visible and hidden edges.
3. Downsample the Mask buffer to half
4. Edge detect the half res buffer, with different colors to hidden and visible edges.
5. Blur the edges of half res/quarter res buffer(quarter res is used for edge glow)
6. Add the blurred edges on top of original scene using the full res mask buffer

lejeunerenard committed:
Add missing intersectSphere() documentation for Ray (#9412)

* Add intersectSphere() method to Ray docs

* Remove ?s in Ray documentation after return values

These are unique phenomenon in the current docs and don't seem to
indicate anything meaningful.

mrdoob committed:
Clean up.

mrdoob committed:
Updated builds.

mrdoob committed:
Moved gaussianPdf out of core.

mrdoob committed:
Updated builds.

lejeunerenard committed:
Fix method link ReturnTypes for math object documenation (#9413)

* Fix method link 'ReturnType's of all math object documentation

* Misc Documentation fixes

AtiX committed:
Adds support for combined Depth+Stencil Texture (#9368)

* Adds THREE.UnsignedInt248Type and THREE.DepthStencilFormat constants

* Extends DepthTexture to allow for either THREE.DepthFormat or THREE.DepthStencilFormat

* Extends depthTexture initialization to support both DepthTextureFormat and DepthStencilFormat

* Fixes typos

* Ensures correct internal format for DepthStencil textures

* Allows to set depthTexture via options.depthTexture in constructor

GGAlanSmithee committed:
Adds scene fog JSON serialization and parsing (#9369)

* adds toJSON methods to scene and fog objects

* adds toJSON method to FogExp2

* adds fog instantiation to ObjectLoader.parse

* adds type info to fog JSON serialization and loading

* fixes unresolved conflict from rebase

mrdoob committed:
WebGLTextures: Modularised Depth*Format.

mrdoob committed:
WebGLRenderer: Fixed missing DepthStencilFormat import.

mrdoob committed:
Updated builds.

satori99 committed:
Parse JSON Material .fog property (#9414)

mrdoob committed:
Added REVISION back.

mrdoob committed:
Updated builds.

Rich-Harris committed:
force Rollup to use correct indentation (#9420)

mrdoob committed:
Updated builds.

genu committed:
Clarify the explanation (#9422)

First usage of `updateStyle` with now indication of what or where it came from.

Commit:
fixing performance when loading very large ASCII STL files (#9449)

WestLangley commented:
Can you please fork and target the `dev` branch, instead? Then, your PR can be considered.

https://github.com/mrdoob/three.js/wiki/How-to-contribute-to-three.js

paulmasson committed:
Grammar corrections (#9453)

takahirox committed:
Clean up VREffect (#9443)

* Clean up VREffect

* Remove the space after function name

takahirox committed:
Simplify and enhance PropertyBinding Regex (#9330)

* Simplify and enhance PropertyBinding Regex

* Update Regex of PropertyBinding.

* Update regular expression of PropertyBinding

takahirox committed:
Remove unnecessary lines of PropertyBinding. (#9362)

tentone committed:
Added depth characteristics to Material JSON serializer (#9448)

mrdoob committed:
Material: Clean up.

mkeblx committed:
StereoEffect set eye separation (#9432)

mrdoob committed:
StereoEffect: setEyeSep() to setEyeSeparation()

mrdoob committed:
Updated builds.

ascloutier committed:
Maya Exporter: Export Selected, Nicer JSON, default extension (#9431)

I’ve added a handful of improvements that I think will really help
people trying to use Maya with 3JS:

1. The current plugin will only perform an “EXPORT ALL” regardless of
what you select from the menu in Maya. I have fixed that so you can now
“Export All” or “Export Selected”. Export all sill merges geometry
rather than exporting a 3JS scene, may fix that in the future.
2. Updated the default extension to be “.json” rather than “.js”. This
is more in-line with 3JS and the other exporters.
3. I pulled the JSON encoder from the Blender exporter and integrated
it into the Maya one so that each item in the ups, faces, vertices, etc
arrays are no longer on a separate line. Makes the file MUCH easier to
read and debug.
4. Turned off sorting of the JSON output. This is more of an opinion,
but I didn’t feel like it helped the output at all.

satori99 committed:
Serialize and load JSON Light .shadow property (#9416)

* Serialize and load JSON Light .shadow property

* Add .toJSON() method to LightShadow
* Light .toJSON() writes .shadow property
* Parse .shadow property in DirectionLight and Spotlight JSON data

* Remove .castShadow condition when exporting .shadow property to JSON

mrdoob committed:
ObjectLoader: Handle data.shadow outside of the switch.

mrdoob committed:
Clean up.

mrdoob committed:
Updated builds.

mrdoob committed:
Somehow this is no longer needed.

milcktoast committed:
Fix Matrix3 Matrix4 getInverse throwOnDegenerate (#9405)

* Fix Matrix3 Matrix4 getInverse throwOnDegenerate

* Update throwOnDegenerate assertion

mrdoob committed:
XHRLoader: Check if overrideMimeType is available before using it. Fixes #9376.

mrdoob committed:
Updated builds.

takahirox committed:
Remove the workaround for Japanese character Bone name in MMDLoader (#9469)

satori99 committed:
ObjectLoader reports image load errors to LoadingManager (#9474)

mrdoob committed:
Clean up.

mrdoob committed:
Updated builds.

yn5 committed:
Add typechecking for array argument in BufferAttribute (#9445)

mrdoob committed:
Clean up.

mrdoob commented:
You should use the `dev` branch as base.
